duckduckgo/iOS
# Support for Light Theme

Task/Issue URL: https://app.asana.com/0/856498667320406/732857702748674
Tech Design URL: https://app.asana.com/0/856498667320406/872957960932374
CC:

**Description**:
Support for Light Theme in the application, like Android app does.

Notable features:
- Experimentation (see test steps).
- Pixels for entering settings and switching themes.
- New launch screen with fixes to logo alignment.
- New "page not found" image. 
- Dark keyboard appearance used when Dark Theme is set (in all cases, not only for experiments).

**Steps to test this PR**:

_Update scenario_
1. Install old version.
2. Update App.

Expectation: No theme toggle present in settings, Dark Theme is set.

_Fresh install: control group_
1. Set "mg" variant in code.
2. Run the app.

Expectation: No theme toggle present in settings, Dark Theme is set.

_Fresh install: experiment: light theme_
1. Set "mh" variant in code.
2. Run the app.

Expectation: Theme toggle present in settings, light theme by default.

_Fresh install: experiment: light theme_
1. Set "mj" variant in code.
2. Run the app.

Expectation: Theme toggle present in settings, dark theme by default.

_Toggle theme setting

1. Open Settings screen.
2. Change theme. 

Expectation: all screens should have appearance matching chosen theme.

_Persisting Theme setting

1. Open Settings screen.
2. Change theme. 
3. Restart the app.

Expectation: all screens should have appearance matching chosen theme.

_Testing about screen text_

1. Open About screen on iPhone.
2. Open About screen on iPad 

Expectation: text should be larger on iPad.

_Testing pixels_

For following actions:

- Open settings from actions menu.
- Open settings from tab switcher.
- Light mode enabled in settings.
- Light mode disabled in settings.

Pixels should be sent.


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
